plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/index.tsx

```diff
@@ -73,6 +86,10 @@ const Edit = ( props: ProductCollectionEditComponentProps ) => {
 				isUsingReferencePreviewMode={ isUsingReferencePreviewMode }
 				location={ location }
 				usesReference={ props.usesReference }
+				isDeletedProductReference={
+					productCollectionUIStateInEditor ===
+					ProductCollectionUIStatesInEditor.DELETED_PRODUCT_REFERENCE
+				}
```

Comment:
I understand we need it to differentiate `ProductPicker` but I'm not a fan of this solution in which we pass the prop to all the other components as well.

One option to overcome this is to extract `sharedProps` to an object and in the switch above, return specific component, rather than assign `Component = ...`.

In other words, it would be:

```
case ProductCollectionUIStatesInEditor.COLLECTION_PICKER:
	return <ProductCollectionPlaceholder {...sharedProps} />;
case ProductCollectionUIStatesInEditor.PRODUCT_REFERENCE_PICKER:
    return <ProductPicker {...sharedProps} />;
case ProductCollectionUIStatesInEditor.DELETED_PRODUCT_REFERENCE:
	return <ProductPicker {...sharedProps}  isDeletedProductReference />;
...
```

I think it would be cleaner and simpler, wdyt? 😅

Reply:
Thank you for the suggestion, Karol! TBH, I have been thinking about this but was avoiding it because I thought it would require a good amount of refactoring 😅 But after reading this feedback, I decided to do it and have made the changes 🙇‍♂️